content/docs/02-guides/03-llama-3_1.mdx

```diff
@@ -0,0 +1,425 @@
+  prompt: 'What is love?',
+});
+```
+
+<Note>
+  Llama 3.1 is available to use via a number of AI SDK providers including
+  [Groq](/providers/ai-sdk-providers/groq), [Amazon
+  Bedrock](/providers/ai-sdk-providers/amazon-bedrock),
+  [Perplexity](/providers/ai-sdk-providers/perplexity),
+  [Fireworks](/providers/ai-sdk-providers/fireworks), and more.
+</Note>
+
+AI SDK Core abstracts away the differences between model providers, allowing you to focus on building great applications. Prefer to use Llama 3.1-70b? the unified interface also means that you can easily switch between models by changing just two lines of code.
+
+```tsx
+import { generateText } from 'ai';
+import { createOpenAI as createGroq } from '@ai-sdk/openai';
+
+const groq = createGroq({
+  baseURL: 'https://api.groq.com/openai/v1',
+  apiKey: process.env.GROQ_API_KEY,
+});
+
+const { text } = await generateText({
+  model: groq('llama-3.1-70b-versatile'),
+  prompt: 'What is love?',
+});
+```
+
+To stream the model's response as it's being generated, update your code snippet to use the [`streamText`](/docs/reference/ai-sdk-core/stream-text) function.
+
+```tsx
+import { streamText } from 'ai';
+import { createOpenAI as createGroq } from '@ai-sdk/openai';
+
+const groq = createGroq({
+  baseURL: 'https://api.groq.com/openai/v1',
+  apiKey: process.env.GROQ_API_KEY,
+});
+
+const { textStream } = await streamText({
+  model: groq('llama-3.1-70b-versatile'),
+  prompt: 'What is love?',
+});
+```
+
+### Generating Structured Data
+
+While text generation can be useful, your usecase will likely call for generating structured data. For example, you might want to extract information from text, classify data, or generate synthetic data. AI SDK Core provides two functions ([`generateObject`](/docs/reference/ai-sdk-core/generate-object) and [`streamObject`](/docs/reference/ai-sdk-core/stream-object)) to generate structured data, allowing you to constrain model outputs to a specific schema.
+
+```tsx
+import { generateObject } from 'ai';
+import { createOpenAI as createGroq } from '@ai-sdk/openai';
+import { z } from 'zod';
+
+const groq = createGroq({
+  baseURL: 'https://api.groq.com/openai/v1',
+  apiKey: process.env.GROQ_API_KEY,
+});
+
+const { object } = await generateObject({
+  model: groq('llama-3.1-70b-versatile'),
+  schema: z.object({
+    recipe: z.object({
+      name: z.string(),
+      ingredients: z.array(z.object({ name: z.string(), amount: z.string() })),
+      steps: z.array(z.string()),
+    }),
+  }),
+  prompt: 'Generate a lasagna recipe.',
+});
+```
+
+This code snippet will generate a type-safe recipe that conforms to the specified schema!
+
+### Tools
+
+While LLMs have incredible generation capabilities, they struggle with discrete tasks (e.g. mathematics) and interacting with the outside world (e.g. getting the weather). The solution: tools, which are like programs that you provide to the model, which it can choose to call as and when it deems necessary.
```

Comment:
```suggestion
While LLMs have incredible generation capabilities, they struggle with discrete tasks (e.g. mathematics) and interacting with the outside world (e.g. getting the weather). The solution: tools, which are like programs that you provide to the model, which it can choose to call as necessary.
```